app/actions/websocket/users.ts

```diff
@@ -120,3 +124,88 @@ export async function handleStatusChangedEvent(serverUrl: string, msg: WebSocket
     const newStatus = msg.data.status;
     setCurrentUserStatus(serverUrl, newStatus);
 }
+
+export async function handleCustomProfileAttributesValuesUpdatedEvent(serverUrl: string, msg: WebSocketMessage) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const {user_id, values} = msg.data;
+        const attributesForDatabase = Object.entries(values).map(([fieldId, value]) => ({
+            id: customProfileAttributeId(fieldId, user_id),
+            field_id: fieldId,
+            user_id,
+            value: value as string,
+        }));
+
+        try {
+            await operator.handleCustomProfileAttributes({
+                attributes: attributesForDatabase,
+                prepareRecordsOnly: false,
+            });
+        } catch (error) {
+            logError('Error handling custom profile attributes values updated event', error);
+        }
+    } catch (error) {
+        logError('Error getting the operator for the custom profile attributes values updated event', error);
+    }
+
+}
+
+export async function handleCustomProfileAttributesFieldUpdatedEvent(serverUrl: string, msg: WebSocketMessage) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const {field} = msg.data;
+
+        try {
+            await operator.handleCustomProfileFields({
+                fields: [field],
+                prepareRecordsOnly: false,
+            });
+        } catch (error) {
+            logError('Error handling custom profile attributes field updated event', error);
+        }
+    } catch (error) {
+        logError('Error getting the operator for the custom profile field updated event', error);
+    }
+
+}
+
+export async function handleCustomProfileAttributesFieldDeletedEvent(serverUrl: string, msg: WebSocketMessage) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const {field_id} = msg.data;
+
+        try {
+            // Delete the field from the database
+            const fieldForDatabase: CustomProfileField = {
+                id: field_id,
+                group_id: '',
+                name: '',
+                type: '',
+                target_id: '',
+                target_type: '',
+                create_at: 0,
+                update_at: 0,
+                delete_at: Date.now(),
+                attrs: {},
+            };
+
+            await operator.handleCustomProfileFields({
+                fields: [fieldForDatabase],
+                prepareRecordsOnly: false,
+            });
+        } catch (error) {
+            logError('Error handling custom profile field deleted event', error);
+        }
+        try {
+            // Also delete any attributes associated with this field
+            await deleteCustomProfileAttributesByFieldId(operator.database, field_id);
```

Comment:
see my previous comment, if we want to keep it like this, which I'm 3/5 on moving to the handler, this should be move to local actions